src/jit/lsrabuild.cpp

```diff
@@ -1611,6 +1611,13 @@ void LinearScan::buildRefPositionsForNode(GenTree* tree, BasicBlock* block, Lsra
             {
                 minRegCountForRef++;
             }
+#if FEATURE_PARTIAL_SIMD_CALLEE_SAVE
+            else if (newRefPosition->refType == RefTypeUpperVectorSaveDef)
+            {
+                minRegCountForRef += 1;
+            }
+#endif // FEATURE_PARTIAL_SIMD_CALLEE_SAVE
```

Comment:
Normally, one would imagine that the handling for these should be in the initial counting loop (i.e. counting these as for other defs), as any registers actually allocated will remain live across the call. However, since these are optional, it will only remain live if it finds a valid register. So one might think that reg-optional defs shouldn't be counted. But that would require the allocation to handle an empty candidate set for the reg-optional case.
So, as this is a stressRegs-only issue, I think I'm fine with this fix, but there should be a comment explaining this (i.e. that, even though the live range extends beyond the call, these don't need to be counted cumulatively.

Reply:
In my test, I use 10 vectors were there are only 2 free registers (xmm5-xmm6), so what will happen all 10 `RefTypeUpperVectorSaveDef` will use `xmm5` to do a spill on the stack. So this xmm5 register does not live across the call in this case, we will see that we allocate it and then split for the next `RefTypeUpperVectorSaveDef`.

Also, I want to note that there is no way not to allocate a register for this vectors, without significant improvements in our `regOptional` or `allocateIfProfitable` mechanisms.

So if you count them for other defs that you will have `minRegCountForRef == 10` for this test case and it is more than the number of registers that are available for this type.

Reply:
How the asm looks like:
```
IN001c: 00010B vextractf128 ymm4, ymm7, 1
IN001d: 000111 vmovupd  xmmword ptr [TEMP_10 rsp+180H], xmm4
IN001e: 00011B vextractf128 ymm4, ymm8, 1
IN001f: 000121 vmovupd  xmmword ptr [TEMP_09 rsp+170H], xmm4
IN0020: 00012B vextractf128 ymm4, ymm9, 1
IN0021: 000131 vmovupd  xmmword ptr [TEMP_08 rsp+160H], xmm4
IN0022: 00013B vextractf128 ymm4, ymm10, 1
IN0023: 000141 vmovupd  xmmword ptr [TEMP_07 rsp+150H], xmm4
IN0024: 00014B call     DevDiv_714266:CallWithoutUsesAndDefs()
IN0025: 000150 vmovupd  xmm4, xmmword ptr [TEMP_07 rsp+150H]
IN0026: 00015A vinsertf128 ymm10, ymm4, 1
IN0027: 000160 vmovupd  xmm4, xmmword ptr [TEMP_08 rsp+160H]
IN0028: 00016A vinsertf128 ymm9, ymm4, 1
IN0029: 000170 vmovupd  xmm4, xmmword ptr [TEMP_09 rsp+170H]
IN002a: 00017A vinsertf128 ymm8, ymm4, 1
IN002b: 000180 vmovupd  xmm4, xmmword ptr [TEMP_10 rsp+180H]
IN002c: 00018A vinsertf128 ymm7, ymm4, 1
```

Reply:
The support for these `RefPosition` types was added prior to adding (slightly) more general support for optional registers. However, they are indeed treated as optional. See https://github.com/dotnet/coreclr/blob/master/src/jit/lsra.cpp#L5557. 
So, this comment is incorrect. I'm happy to see a comment that says something like "TODO-Cleanup: The register allocator currently treats these as reg-optional, but not in a way that tolerates an empty candidate set."

Reply:
I have tried to mark them as `AllocateIfProfitable` but there are several missing pieces there.

Reply:
Right. My point was that these are not handled using the current mechanism for optional registers. Doing so would be problematic because it's not truly optional. Rather, if no register is found by the initial call to `tryAllocateFreeReg()` it calls it again using all the float regs, including the caller save. It's a hack, because these are always at a call site where it is known that the caller-save registers will be available.
However, it would be perfectly reasonable for `tryAllocateFreeReg()` to simply return `REG_NA` if the register set was empty, and then the change you've made would not be needed.